test/benchmarks/rcv1_benchmarks.cc

```diff
@@ -0,0 +1,278 @@
+1 |f 14:4.6685375e-02 17:3.9043721e-02 24:3.5642762e-02 39:4.6711642e-02 41:6.4986050e-02 53:9.7434938e-02 70:6.0479779e-02 101:6.8327434e-02 125:6.5960057e-02 135:7.7151470e-02 138:6.6490836e-02 140:5.6129370e-02 157:1.8222515e-01 167:7.3569357e-02 183:5.5437617e-02 188:6.1004069e-02 210:5.6579288e-02 229:7.6725185e-02 234:5.5541933e-02 247:8.0911487e-02 257:8.7341174e-02 284:4.4998646e-02 286:5.6236453e-02 293:4.2629275e-02 295:6.2220700e-02 320:6.0314614e-02 326:5.2210093e-02 338:5.5483330e-02 342:7.0478953e-02 394:7.6991834e-02 407:1.4337897e-01 433:4.7747806e-02 453:6.0631219e-02 464:7.1966082e-02 496:1.3877112e-01 516:7.3230974e-02 533:4.1105788e-02 549:6.4592525e-02 613:1.2657496e-01 616:6.8928547e-02 655:4.7937904e-02 684:7.3886104e-02 690:1.5935852e-01 740:2.9913169e-01 760:9.3864232e-02 767:7.1026899e-02 782:1.5962097e-01 812:1.6494313e-01 814:7.1446441e-02 815:7.8162819e-02 844:1.2304020e-01 992:1.2588154e-01 1036:7.5061359e-02 1073:1.7234431e-01 1115:8.0682985e-02 1119:9.9624366e-02 1122:5.3169947e-02 1213:4.4482388e-02 1535:2.1814872e-01 1617:8.4668003e-02 1654:5.0346255e-02 1682:8.2505457e-02 1702:9.2321545e-02 1729:1.2025382e-01 2104:9.1611803e-02 2298:2.9297344e-02 2548:9.2505068e-02 2623:1.2080402e-01 2880:1.2729374e-01 2891:1.1095203e-01 3589:1.3369514e-01 4470:1.2805253e-01 5874:1.1226498e-01 6074:2.0972548e-01 7761:1.2549773e-01 22172:2.2821054e-01 24843:4.1516986e-01
+-1 |f 10:7.0070818e-02 13:3.0282617e-02 24:1.5686616e-02 26:4.1215312e-02 42:6.1233964e-02 55:6.4943276e-02 74:4.0680200e-02 90:8.7173775e-02 98:4.9359940e-02 125:1.0314921e-01 131:5.5968020e-02 133:5.5713881e-02 147:4.0348452e-02 152:5.8046080e-02 161:3.9000869e-02 166:1.0284418e-01 179:3.4781855e-02 188:2.1661038e-02 189:4.0627711e-02 210:4.2160925e-02 219:8.0608368e-02 228:6.6707455e-02 229:3.3767268e-02 231:4.2642169e-02 240:3.6020506e-02 25
```

Comment:
does the memory allocation need to be in the state loop?

Reply:
This needs to be done for each instance of vw, doing it outside is not really possible. This benchmark's goal is to represent an end to end run with the disk IO removed